marlowe-dashboard-client/src/Dashboard/State.purs

```diff
@@ -253,24 +254,32 @@ handleAction input@{ currentSlot } (UpdateContract followerAppId contractHistory
 -- redeemed again anyway the next time the user picks up the wallet. Since these duplicate requests
 -- will happen anyway, therefore, this extra complication doesn't seem worth the bother.
 handleAction input (RedeemPayments followerAppId) = do
-  walletDetails <- use _walletDetails
-  contracts <- use _contracts
-  for_ (lookup followerAppId contracts) \{ executionState, mMarloweParams, userParties } ->
-    for_ mMarloweParams \marloweParams ->
-      let
-        payments = getAllPayments executionState
-
-        isToParty party (Payment _ payee _) = case payee of
-          Party p -> p == party
-          _ -> false
-      in
-        for (List.fromFoldable userParties) \party ->
+  lastMarloweAppEndpointCall <- use _lastMarloweAppEndpointCall
+  case lastMarloweAppEndpointCall of
```

Comment:
I think there's some misunderstanding here. Sorry, I probably should have explained more in the initial PR comments. But this isn't meant to do anything about the multiple redeem problem. It isn't meant to fix any bug at all in fact - it's just fleshing out the handling of the MarloweApp errors.

Since we've currently no way of mapping errors back to the calls that triggered them, I thought the simplest solution for now would just be to remember the last call, and prevent any further ones from happening until we hear back.

I agree it would be better ultimately to have request IDs that show up in the MarloweApp state along with the error/ok message. Mind you, now that I think about it this would have to be an array/map of all the requests sent to the app, so after a while it would get pretty long and potentially start to slow things down. We might need to think about it a bit more.

Reply:
If this isn't solving a critical bug, then I'm not in favour of this change going in even as a temporary one.

Reply:
It's not solving a critical bug, but it is adding critical functionality. We need to handle MarloweApp errors properly. And to do that, we need to be able to match them up with the calls that triggered them.

I really don't think this solution is that bad. Sure, it's not ideal. But it works, and can be implemented now, in time for the September demo. I can't see another option that isn't going to take quite a bit more work (and work on the backend code).